tools/ci-build.sh

```diff
@@ -5,6 +5,9 @@
 set -e  # Exit on error.
 set -x  # Make command execution verbose
 
+THIS_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+P4C_DIR=$(readlink -f ${THIS_DIR}/..)
```

Comment:
readlink -f $0/..

Reply:
IIRC `readlink -f` is not available in macos

Reply:
I think the other version is more readable, any reason why `$0`?

Reply:
$0 is part of the POSIX shell specification, whereas BASH_SOURCE, as the name suggests, is Bash-specific.